### PR TITLE
docs: add GitHub Actions guidance

### PR DESCRIPTION
**Pull request by kanadgupta**
This is a doozy. This PR updates our README.md, adds docs for [our knowledge base](https://docs.readme.com/docs/rdme), and sets up a CI workflow for doing a find/replace and syncing the knowledge base page to ReadMe.

Fixes RM-3553

### CODE REVIEW

File: README.md
Comment (RyanGWU82) [inline README.md:15]: Why does `devDependencies` prevent "unexpected behavior with mismatching versions"? (Is there a clearer reason why to recommend using `devDependencies`, like "because you don't need it in running on your production servers"?)

File: README.md
Comment (RyanGWU82) [inline README.md:76]: In this example, could you show to use a GitHub Actions secret in place of the variables?

**Review by RyanGWU82 — APPROVED** [review]
Very nice! 👏🏼

File: README.md
Comment (kanadgupta) [reply README.md:15]: Ah that's a great callout. The reasoning for `rdme` being installed in `devDeps` is separate from the reasoning for mismatching versions... I'll clarify!

File: README.md
Comment (kanadgupta) [reply README.md:76]: I thought about it but I opted not to because I was worried it would make the illustration of the usage harder to follow. I was thinking of having a lengthier section on using GitHub secrets in [our dedicated docs page](https://docs.readme.com/docs/automatically-sync-api-specification-with-github). Thoughts?

Comment (RyanGWU82) [reply README.md:76]: ¯\_(ツ)_/¯ That sounds useful but also seems a bit beyond the scope of our docs page. You could probably keep it simple and just link to here: https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow

Comment (kanadgupta) [reply README.md:76]: Oh yeah to clarify, "lengthier" is doing a lot of work in that last comment lol... I was planning on linking to that GitHub doc page and just including another example that uses secrets!

File: README.md
Comment (erunion) [inline README.md:7]:
```suggestion
`rdme` is the CLI and [GitHub Action](https://docs.github.com/actions) wrapper for [ReadMe's RESTful API](https://docs.readme.com/reference/intro-to-the-readme-api). It allows you to sync [OpenAPI](https://spec.openapis.org) and [Swagger](https://swagger.io/specification/v2/) definitions with projects you create on [ReadMe](https://readme.com/). Additionally, you can sync documentation with your project, and manage project versions.
```

**Review by erunion — APPROVED** [review]

File: README.md
Comment (kanadgupta) [inline README.md:71]: Just discovered this command which is incorrect since the `--id` flag implies the project version 🤦🏽

File: README.md
Comment (erunion) [inline README.md:52]: still need this?

File: documentation/rdme.md
Comment (erunion) [inline documentation/rdme.md:66]: Marketplace listing link is missing. And should I publish a 7.0 release once this is merged?

Comment (kanadgupta) [reply documentation/rdme.md:66]: I have a TODO for that here: https://github.com/readmeio/rdme/blob/c4bd19851f14a45cb74bd9fe5c75c65dc2db5085/documentation/rdme.md#L62

I keep going back-and-forth on the version. How about we ship v6.5 once this is out, and then merge in #210 and immediately ship v7?

File: README.md
Comment (kanadgupta) [reply README.md:52]: nope, thank you!

File: documentation/rdme.md
Comment (erunion) [reply documentation/rdme.md:66]: sounds good to me

File: README.md
Comment (domharrington) [inline README.md:7]: Is https://spec.openapis.org the best place to link to for this? It looks bare and pretty ugly:

image

File: README.md
Comment (domharrington) [inline README.md:50]: Do you also have to install `rdme` in devDeps if you're using the github action, or no?

File: documentation/rdme.md
Comment (domharrington) [inline documentation/rdme.md:10]:
```suggestion
This Markdown page is syncing to ReadMe via the `rdme` GitHub Action 🦉
```

**Review by domharrington — APPROVED** [review]

File: README.md
Comment (kanadgupta) [reply README.md:50]: Nope! You just list it in your workflow YAML file and you're good to go 🚀 

That's also why I have this callout here so people don't install the package for GitHub Action usage:https://github.com/readmeio/rdme/blob/9e895ffc69504b373a9d5593c6d44219d52b479e/README.md#L13

File: README.md
Comment (kanadgupta) [reply README.md:7]: Ugh yeah so I took [this link](https://spec.openapis.org/oas/v3.1.0.html) and dropped most of the params so people could discover their own spec version, but agreed, that landing page is hideous and not very intuitive.

I think it was Jon who discovered that site recently and I think I like linking to [this](https://spec.openapis.org/oas/v3.1.0.html) as opposed to the giant Markdown file... maybe I'll just include the full spec file link here instead of that ugly landing page? Any thoughts on what this link should be @readmeio/api?

Comment (erunion) [reply README.md:7]: I like that site you linked @kanadgupta

**Comment by kanadgupta** [comment]
@erunion @RyanGWU82 ready for review again. The docs are pretty long and have been mostly signed off on for all-intensive purposes, so we can just make any copy edits in upcoming PRs after we get `v6.5` released. The main thing I want y'alls blessing on is the CI workflow I set up in these files:
- https://github.com/readmeio/rdme/blob/docs/github-action/.github/workflows/docs.yml
- https://github.com/readmeio/rdme/blob/docs/github-action/bin/set-version-output
- https://github.com/readmeio/rdme/blob/docs/github-action/src/lib/getNodeVersion.js (see [here](https://github.com/readmeio/rdme/blob/docs/github-action/__tests__/lib/getNodeVersion.test.js) for corresponding tests)

File: .github/workflows/docs.yml
Comment (erunion) [inline .github/workflows/docs.yml:54]: You've probably already made a note of this but we'll probably need to update this when Ray's docs project gets underway.

File: .github/workflows/docs.yml
Comment (erunion) [inline .github/workflows/docs.yml:48]:
```suggestion
        # So we can owlfood our Github action we use the `main` branch as a ref
```

Comment (erunion) [reply .github/workflows/docs.yml:48]: please dont accept this

File: src/lib/getNodeVersion.js
Comment (erunion) [inline src/lib/getNodeVersion.js:5]: Not sure if you want the min or max, but this code currently returns the latter:

```js
> Array.from("^12 || ^14 || ^16".matchAll(/\d+/g)).pop();
['16', index: 15, input: '^12 || ^14 || ^16', groups: undefined]
```

```suggestion
 * @returns {String} The maximum major Node.js version specified in the package.json
```

**Review by erunion — APPROVED** [review]

File: src/lib/getNodeVersion.js
Comment (kanadgupta) [reply src/lib/getNodeVersion.js:5]: Ooops thanks... yeah I wrote this comment when I was using `semver.minVersion` but ended up scrapping that since that'd require installing `semver` in the workflow 🙄

**Pull request merged**
